src/js/helpers/contractGrantActivityHelper.js

```diff
@@ -205,6 +204,10 @@ export const getXDomain = (dates, awardType, transactions) => {
             const date = !badEnd ? potentialEndDate : currentEndDate;
             return [transactions[0].action_date.valueOf(), date.valueOf()];
         }
+        if (!badStart && (!badCurrent || !badEnd)) {
```

Comment:
these kinds of confusing conditionals it really helps me to use a variable so I can name it. Confusing stuff going on here kinda; especially with the ternary on 208. Looks like we're mostly checking for the case where the potential End date is in the future but the end date is in the past or something...? Hard to tell.